### ISSUE TITLE
🐛 `useExhaustiveDependencies`: false positive for out of scope dependency

### ISSUE THREAD

**Issue by emab** [biomejs/biome#651]
### Environment information

```block
CLI:
  Version:                      1.3.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "cygwin"
  JS_RUNTIME_VERSION:           "v18.12.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0
```

### What happened?

When using a variable declared outside of the component scope, you should not add it to a React `useEffect` dependency array. A valid example of this in the React docs:

- https://react.dev/learn/you-might-not-need-an-effect#initializing-the-application

ESLint is aware of this, and gives a warning if it is added:

> ESLint: React Hook useEffect has an unnecessary dependency: 'hasRequested'. Either exclude it or remove the dependency array. Outer scope values like 'hasRequested' aren't valid dependencies because mutating them doesn't re-render the component.(react-hooks/exhaustive-deps)

However Biome falsely tells you to add it as a dependency:

https://biomejs.dev/playground/?lintRules=all&code=aQBtAHAAbwByAHQAIAB7AHUAcwBlAEUAZgBmAGUAYwB0AH0AIABmAHIAbwBtACAAJwByAGUAYQBjAHQAJwA7AAoACgBsAGUAdAAgAGQAaQBkAEkAbgBpAHQAIAA9ACAAZgBhAGwAcwBlADsACgAKAGYAdQBuAGMAdABpAG8AbgAgAEEAcABwACgAKQAgAHsACgAgACAAdQBzAGUARQBmAGYAZQBjAHQAKAAoACkAIAA9AD4AIAB7AAoAIAAgACAAIABpAGYAIAAoACEAZABpAGQASQBuAGkAdAApACAAewAKACAAIAAgACAAIAAgAGQAaQBkAEkAbgBpAHQAIAA9ACAAdAByAHUAZQA7AAoAIAAgACAAIAAgACAALwAvACAABScgAE8AbgBsAHkAIAByAHUAbgBzACAAbwBuAGMAZQAgAHAAZQByACAAYQBwAHAAIABsAG8AYQBkAAoAIAAgACAAIAB9AAoAIAAgAH0ALAAgAFsAXQApADsACgAgACAALwAvACAALgAuAC4ACgB9AA%3D%3D

### Expected result

You should not be prompted to add an out-of-scope variable to the dependency array.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Duplicate of https://github.com/biomejs/biome/issues/608

**Comment by emab** [comment]
> Duplicate of #608

I don't think this is a duplicate of that issue?

**Comment by ematipico** [comment]
Apologies!

**Comment by kalleep** [comment]
I can pick this one up
